core/src/test/java/io/opencensus/stats/MeasureTest.java

```diff
@@ -110,6 +110,12 @@ public void testLongMeasureEquals() {
         .testEquals();
   }
 
+  @Test
+  public void testDoubleMeasureIsNotEqualToLongMeasure() {
+    assertThat(Measure.DoubleMeasure.create("name", "description", "bit/s"))
+        .isNotEqualTo(Measure.LongMeasure.create("name", "description 2", "bit/s"));
```

Comment:
I think it would be better to compare two Measures with the same description.  Even if we decide that recording Measures with the same name is equivalent in `StatsRecorder.record`, `Measure.equal` should compare all fields and the class.